js/api.ts

```diff
@@ -133,8 +133,8 @@ export function fetchFirestoreJWT(classHash: string, firebaseApp?: string) {
 }
 
 export function authFirestore(rawFirestoreJWT: string) {
-  return signInWithToken(rawFirestoreJWT).catch(err => {
-    // tslint:disable-next-line no-console
+  const authResult = signInWithToken(rawFirestoreJWT) as Promise<any>;
```

Comment:
I think this should be Promise<firebase.auth.UserCredential | void> if we want to be fully correct, but it doesn't matter as we don't use return value anyway. I'm surprised it requires manual casting tho. Why did you have to change these lines and cast type? Was it ESLint thing?

Reply:
Newer version of TypeScript complained about incompatible types. I think they tightened up some of their type-checking around promises somewhere along the way. As I recall I tracked it down to TS 3.6. I'll try what you suggest because that would be a better workaround.

Reply:
FYI, without the cast the error is:
```
ERROR in /Users/kswenson/Development/cc-dev/portal-report/js/api.ts
ERROR in /Users/kswenson/Development/cc-dev/portal-report/js/api.ts(137,21):
TS2349: This expression is not callable.
  Each member of the union type '(<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<TResult>) | null | undefined) => Promise<void | TResult>) | (<TResult = never>(onrejected?: ((reason: any) => TResult | PromiseLike<...>) | ... 1 more ... | undefined) => Promise<...>)' has signatures, but none of those signatures are compatible with each other.
```

Reply:
Excellent! The cast to `Promise<firebase.auth.UserCredential | void>` you suggested worked so I'll make that change. I thought I had tried that variant but apparently not.